Tools/Scripts/webkitdirs.pm

```diff
@@ -617,16 +628,46 @@ sub determineXcodeDestination
     }
         
     if (!$generic && $xcodeSDKPlatformName =~ /simulator$/) {
+        # Two goals:
+        # 1. Find a simulator device to build for, to avoid building multiple architectures.
+        # 2. Try to pick a simulator that's been used before (either by command-line or IDE builds) to avoid
+        #    unnecessary recompilation.
+        #
+        # Changing the targeted simulator between builds breaks incremental building, because it changes
+        # build settings like TARGET_DEVICE_IDENTIFIER.
+        my $prevBuildRequest = first {
+            $_->{parameters}->{activeRunDestination}->{platform} eq $xcodeSDKPlatformName
+        } xcodeBuildRequestsInRecencyOrder();
+        my $prevUDID = $prevBuildRequest->{parameters}->{overrides}->{synthesized}->{table}->{TARGET_DEVICE_IDENTIFIER} if $prevBuildRequest;
+        if ($prevBuildRequest && !$prevUDID) {
+            warn "Can't find UDID in previous $xcodeSDKPlatformName build request, builds may not be incremental.\n";
+        }
+        
+        my @devices = iOSSimulatorDevices();
+        my $prevDevice = first { $prevUDID && $_->{UDID} eq $prevUDID } @devices;
+        if ($prevUDID && !$prevDevice) {
+            warn "Simulator with UDID '$prevUDID' not found, falling back to another available simulator. " .
+                "This build may not be incremental.\n";
+        }
+        
+        # If we found the previous device, check that the runtime being built has not changed (e.g. due to a
+        # major SDK update). If it has changed, or if no previous device is available, fall back to the first
+        # eligible device in the list (matching legacy behavior).
```

Comment:
How does Xcode decide which device to build for? It would be best to match it, not legacy behavior.

I'm thinking of the case of building with `make` first, and then with Xcode. It would be best to avoid rebuilding in this case, too.

Reply:
Xcode doesn't decide, you have to pick a destination. Though the list in the UI appears to be in reverse order of what `xcrun simctl list devices` shows. Perhaps we should default to the _last_ eligible device, so that it's akin to picking the first device in the UI. And perhaps we should print the device name we're using so that power users can pick it manually.

Reply:
> Though the list in the UI appears to be in reverse order of what `xcrun simctl list devices` shows.

It's actually lexicographic; the patch I'm landing does this too.